packages/hardhat-core/src/internal/hardhat-network/provider/fork/ForkBlockchain.ts

```diff
@@ -247,16 +247,25 @@ export class ForkBlockchain
       rpcBlock === null ||
       rpcBlock.hash === null ||
       rpcBlock.number === null ||
-      rpcBlock.number.gt(this._forkBlockNumber)
+      rpcBlock.number > this._forkBlockNumber
     ) {
       return undefined;
     }
 
+    const forkNetworkId = this._jsonRpcClient.getNetworkId();
+    let common: Common;
+    if (Chain[forkNetworkId] !== undefined) {
+      common = new Common({ chain: forkNetworkId });
+    } else {
+      // unknown networks use Goerli's common to prevent the
+      // maxExtraDataSize check
+      common = new Common({ chain: 5 });
```

Comment:
won't this break everything? is this used anywhere during the execution?

Reply:
This is a temporary fix, it has a proper fix in a later PR.